src/metpy/plots/declarative.py

```diff
@@ -1986,3 +1988,273 @@ def _build(self):
+    COLD_color.__doc__ = """Color for plotting cold fronts.
+    
+    A single string (color name or hex code) used to plot cold fronts. Default
+    color is 'blue', which is used by `ColdFront()` class. `COLD_color` alternates
+    with `WARM_color` to plot stationary fronts.
+    """
+
+    OCFNT_color = Union([Unicode()], default_value='purple', allow_none=True)
+    OCFNT_color.__doc__ = """Color for plotting occluded fronts.
+    
+    A single string (color name or hex code) used to plot Occluded fronts. Default
+    color is 'purple', which is used by `OccludedFront()` class.
+    """
+
+    TROF_color = Union([Unicode()], default_value='darkorange', allow_none=True)
+    TROF_color.__doc__ = """Color for plotting trough lines.
+    
+    A single string (color name or hex code) used to plot trough lines. Default
+    color is 'darkorange'.
+    """
+
+    HIGH_label = Union([Unicode()], default_value='H', allow_none=True)
+    HIGH_label.__doc__ = """Label used to plot high-pressure systems.
+
+    Single string used as marker to plot high-pressure systems. Default value is 'H'.
+    """
+
+    LOW_label = Union([Unicode()], default_value='L', allow_none=True)
+    LOW_label.__doc__ = """Label used to plot low-pressure systems.
+
+    Single string used as marker to plot low-pressure systems. Default value is 'L'.
+    """
+
+    TROF_linestyle = Union([Unicode()], default_value='dashed',
+                   allow_none=True)
+    TROF_linestyle.__doc__ = """Linestyle of Trough lines.
+
+    Single string, default value is 'dashed'. 
+    Accept matplotlib linestyles: 'solid', 'dotted', 'dashdot'.
+    """
+
+    label_fontsize =  Union([Int(), Float(), Unicode()], default_value=10, allow_none=True)
+    label_fontsize.__doc__ = """Font sizes of pressure systems labels.
+
+    Accepts size in points or relative size. Allowed relative sizes are those of Matplotlib:
+    'xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large'.
+    """
+
+    TROF_linewidth = Union([Float()], default_value=2,
+                         allow_none=True)
+    TROF_linewidth.__doc__ = """Stroke width for trough lines.
+
+    A single integer or floating point value representing the size of the stroke width. 
+    """
+
+    FRONT_linewidth = Union([Float()], default_value=1,
+                         allow_none=True)
+    TROF_linewidth.__doc__ = """Stroke width for front lines.
+
+    A single floating point value representing the size of the stroke width. 
+    """
+
+    FRONT_marker_size = Union([Int(), Float(), Unicode()], default_value=3, allow_none=True)
+    FRONT_marker_size.__doc__ = """Size of symbols in front lines.
+
+    Accepts size in points or relative size. Default value is 3. Allowed relative sizes are those 
+    of Matplotlib: 'xx-small', 'x-small', 'small', 'medium', 'large', 'x-large', 'xx-large'.
+    """
+
+    strength_offset = Union([Tuple()], default_value=(0,-6), allow_none=True)
```

Comment:
Let's make the default offset `(0, -1)` and scale automatically to fit with `StationPlot`.